corehq/warehouse/models/dimensions.py

```diff
@@ -1,33 +1,83 @@
-from django.db import models
-
-
-class BaseDim(models.Model):
+from django.db import models, transaction, connections
+
+from corehq.warehouse.const import (
+    USER_DIM_SLUG,
+    GROUP_DIM_SLUG,
+    LOCATION_DIM_SLUG,
+    DOMAIN_DIM_SLUG,
+    USER_LOCATION_DIM_SLUG,
+    USER_GROUP_DIM_SLUG,
+    USER_STAGING_SLUG,
+    GROUP_STAGING_SLUG,
+    LOCATION_STAGING_SLUG,
+    DOMAIN_STAGING_SLUG,
+)
+
+from corehq.sql_db.routers import db_for_read_write
+from corehq.util.test_utils import unit_testing_only
+from corehq.warehouse.etl import CustomSQLETLMixin
+from corehq.warehouse.models.shared import WarehouseTableMixin
+
+
+class BaseDim(models.Model, WarehouseTableMixin):
```

Comment:
we can handle in another pr but we should probably track batch_id on all our tables